File: SchoolWebApp-Client/src/app/academics/models/education-level-subject.ts
import { ResourceModel } from "@/core/models/ResourceModel";
import { AcademicYear } from "@/school/models/academic-year";
import { Subject } from "./subject";
import { EducationLevel } from "@/school/models/educationLevel";

export class EducationLevelSubject extends ResourceModel<EducationLevelSubject> {
    public educationLevelId: number;
    public educationLevel?: EducationLevel;

    public subjectId: number;
    public subject?: Subject;

    public academicYearId: number;
    public academicYear?: AcademicYear;

    public description?: string;

    constructor(model?: Partial<EducationLevelSubject>) {
        super(model);
      }
}

File: SchoolWebApp-API/Project.Core/Entities/Academics/EducationLevelSubject.cs
using SchoolWebApp.Core.Entities.School;
using SchoolWebApp.Core.Entities.Shared;

namespace SchoolWebApp.Core.Entities.Academics
{
    public class EducationLevelSubject : Base
    {
        public string? Description { get; set; }
        public int EducationLevelId { get; set; }
        public EducationLevel? EducationLevel { get; set; }
        public int SubjectId { get; set; }
        public Subject? Subject { get; set; }
        public int AcademicYearId { get; set; }
        public AcademicYear? AcademicYear { get; set; }
    }
}
